### DISCUSSION TITLE
Trying to set the number of hours of a target session based on the number of periods under a certain price point

### DISCUSSION THREAD

**Discussion by djdarlo** [BottlecapDave/HomeAssistant-OctopusEnergy#746] [Q&A]
I want to set the number of hours in a target session depending on how many 30 min periods are under a certain price point.  e.g. if there are 6 slots under 10p then I want to set the number of hours to 3 (6*0.5).  I have got a basic loop working but the increment counter within the loop isn't increasing which I thought I'd solved previously by moving to an array.

{% if state_attr('event.octopus_energy_electricity_XXX_XXX_current_day_rates', 'rates') | length > 0 %}
Next Target Times
{% set matches = [] %}

{% for ss in state_attr('event.octopus_energy_electricity_XXX_XXX_current_day_rates', 'rates') -%}
{% if (ss.start.strftime('%H')) >= '16' and (ss.value_inc_vat) < 0.30 %}
Time: {{ (ss.start.time()) }}
Hour: {{ ss.start.strftime('%H') }}
Date: {{ (ss.start) }} to {{ (ss.end) }}
Cheap Today {{ (ss.value_inc_vat) }}

{# Create a new list with the match indicator for each iteration #}
{% set matches = matches + [ss.value_inc_vat] %}
Matches: {{ matches }}
{%- endif %}
{%- endfor %}

{%- else -%}
{%- endif %}

Can anyone help me with the logic in my loop code or better still wrap this into code which will set the hours in the target set?

**Comment by SteveGBuck** [comment]
In case it helps. I do something somewhat similar to manage ev charging on agile but with a slightly different approach to yourself.

I have a target rate sensor which is dynamically set based on the hours needed to charge the car.

I also created a helper that allows me to set a pence per kWh limit.

I then have an automation that has a couple of triggers:

- When the target rate sensor switches on
- when the current rate sensor drops below my helper limit.

And some conditions:

- target rate sensor is on
- current rate must be below my helper limit
- Car must be at home


The action is to charge the car.


I have a second automation which is the opposite of the above to stop charging when the rate increases above my helper value or the target rate sensor turns off.
